src/Services/GraphQL.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace BoldApps\ShopifyToolkit\Services;
+
+class GraphQL extends base
+{
+    public function query($data)
```

Comment:
I think it would be nicer to separate the query string and variables in this signature and then build the body for the request within.

```suggestion
    public function query($query, $variables = null)
```

Then the body becomes

```php
return $this->client->post(
  "{$this->getApiBasePath()}/graphql.json",
  [],
  ['query' => $query, 'variables' => $variables],
  [],
  null,
  false,
  ['Content-Type' => 'application/json']
);
```

and calls can be made with variables:

```php
$graphQLService->query(<<<QUERY
mutation sellingPlanGroupDelete($id: ID!) {
    sellingPlanGroupDelete(id: $id) {
        deletedSellingPlanGroupId
        userErrors {
            field
            message
        }
    }
}
QUERY, ['id' => 'gid://shopify/SellingPlanGroup/1234']);
```

I'm also not sure the content type change is necessary. I've always used json.

Reply:
Updated 👍